### PR TITLE
Fix :meth:`~.input_to_graph_point` when passing a `line_graph`

### PR DESCRIPTION
**Pull request by leleogere**
## Overview: What does this pull request change?

Fix a bug preventing to pass a graph returned by `ax.get_line_graph(...)["line_graph"]` from being passed to `ax.i2gp(x, line_graph)`

Get point from `line_graph`

## Further Information and Comments

I typed the `graph` parameter as `Union["ParametricFunction", VMobject]` as `.get_line_graph(...)["line_graph"]` returns a VMobject but it sounds a bit too general. I don't know what else to do tho, I'm open so suggestions!

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/mobject/coordinate_systems.py [hunk 1]
```diff
@@ -743,13 +751,17 @@ def construct(self):
                     0
                 ],
                 target=x,
-                lower_bound=self.x_range[0],
-                upper_bound=self.x_range[1],
+                lower_bound=0,
+                upper_bound=1,
```
Comment (huguesdevimeux) [inline manim/mobject/coordinate_systems.py:755]: **Question :** Why did you change it to 0 and 1 ?

**Review by huguesdevimeux — CHANGES_REQUESTED** [review]
just a very little thing, but overall SGTM. 

Thanks !

File: manim/utils/simple_functions.py [hunk 2]
```diff
@@ -95,9 +95,9 @@ def binary_search(function, target, lower_bound, upper_bound, tolerance=1e-4):
         mh = np.mean([lh, rh])
         lx, mx, rx = (function(h) for h in (lh, mh, rh))
         if lx == target:
-            return lx
+            return lh
         if rx == target:
-            return rx
+            return rh
```
[the remark this answers is not in the document]
Comment (leleogere) [reply manim/utils/simple_functions.py:100]: It took me a while to figure out what the problem was tbh! 😂

File: manim/mobject/coordinate_systems.py (hunk 1 above)
Comment (leleogere) [reply manim/mobject/coordinate_systems.py:755]: We are looking for an alpha value to use in `point_from_proportion`, so it needs to be between 0 and 1, not between `x_range[0]` and `x_range[1]`

Comment (huguesdevimeux) [reply manim/mobject/coordinate_systems.py:755]: You're right. Thanks for the clarification.

**Review by huguesdevimeux — APPROVED** [review]

**Review by hydrobeam — APPROVED** [review]
The changes LGTM 🎉 Thank you for fixing the bug!

File: manim/mobject/coordinate_systems.py [hunk 3]
```diff
@@ -740,13 +755,19 @@ def construct(self):
...
             )
             if alpha is not None:
                 return graph.point_from_proportion(alpha)
             else:
-                return None
+                raise ValueError(
+                    "x={} not located in the range of the graph ([{}, {}])".format(
+                        x,
+                        self.p2c(graph.get_start())[0],
+                        self.p2c(graph.get_end())[0],
```
Comment (hydrobeam) [inline manim/mobject/coordinate_systems.py:768]: Actually I missed one thing😅:

nitpicky, but I think this should be an f-string to be more consistent with the rest of the library.

Comment (leleogere) [reply manim/mobject/coordinate_systems.py:768]: You mean something like that?
```py
f"x={x} not located in the range of the graph ([{self.p2c(graph.get_start())[0]}, {self.p2c(graph.get_start())[0]}])" 
```

Comment (hydrobeam) [reply manim/mobject/coordinate_systems.py:768]: Yup, like that👍

Comment (leleogere) [reply manim/mobject/coordinate_systems.py]: This is fixed!
